File: server/services/battle/battle_field.dart
import 'battle.dart';

class BattleField {
  const BattleField({
    required this.tribe,
    required this.population,
    this.def = 0,
    this.party = false,
    this.durBonus = 1,
    this.wall = const Wall(0, 1),
  });

  final int tribe;
  final int population;
  final int durBonus;
  final Wall wall;
  final int def;
  final bool party;
}

File: java_server/src/main/java/xyz/qruto/java_server/models/battle/BattleField.java
package xyz.qruto.java_server.models.battle;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class BattleField {
    private int tribe;
    private int population;
    @Builder.Default
    private int durBonus = 1;
    @Builder.Default
    private Wall wall = new Wall(0, 1);
    private int def;
    private boolean party;
}
